src/worker.cc

```diff
@@ -472,20 +492,24 @@ void parse_locations(const rapidjson::Document& doc,
     auto request_locations =
         rapidjson::get_optional<rapidjson::Value::ConstArray>(doc, std::string("/" + node).c_str());
     if (request_locations) {
+      uint32_t last_loc_idx = request_locations->Size() - 1;
       for (const auto& r_loc : *request_locations) {
         auto* loc = locations->Add();
-        loc->mutable_correlation()->set_original_index(locations->size() - 1);
-        parse_location(loc, r_loc, options, ignore_closures);
+        auto loc_idx = locations->size() - 1;
+        loc->mutable_correlation()->set_original_index(loc_idx);
+        parse_location(loc, r_loc, request, ignore_closures, loc_idx == last_loc_idx);
```

Comment:
We need to pass to `parse_location` the info whether this location is the last one.